Upstream merge fixes

chris-j-h opened the pull request:
Fixes relating to recent upstream geth merges.  

* https://github.com/ethereum/go-ethereum/pull/21509
    * The upstream merge introduced `ActivePrecompiles()` however there is a typo causing tests added in this PR to fail.  This bug was fixed in upstream 1.10.2 - the content of that fix has been included in this change.  To simplify future merges the content of the commit has been copied instead of cherry-picking the commit.
    * The Quorum Privacy Precompile address has been added to `ActivePrecompiles()` list. This is currently only used for the experimental YoloV2 EIP 2929 (where all active precompiles for the current block height must be known) so not critical but worth fixing ahead of the concrete release that will be pulled in from a future upstream geth merge.

Krish1979 reviewed (dismissed):
LGTM

antonydenyer reviewed (changes requested):
I don't think cherry-picking upstream changes makes sense. We should endeavour to stay up to date with geth changes. Pulling in an additional changes that aren't part of geth v1.9.25 makes it difficult to track and maintain what is going on with quorum.

chris-j-h committed:
Add Quorum Privacy Precompile to ActivePrecompiles list
This is currently only used for the experimental YoloV2 EIP 2929 (where all active precompiles for the current block height must be known) so not critical but worth fixing ahead of the concrete release that will be pulled in from a future upstream geth merge

Also includes core/vm: fix Byzantium address list (#22603) (cherry picked from commit 44fe466999ce7f8b02de7e83de955f8847342309)

Krish1979 reviewed (approved):
LGTM
